### PR TITLE
plugin.video.aod 0.4.13

### PR DESCRIPTION
**Pull request by tyl0re**
### Description

New Plugin, SHows Videos from https://www.anime-on-demand.de/
A Login Is Needed. The First Episode is Free, For Every Else you need an Abo

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ X I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [ X Each add-on submission should be a sinlge commit with using the following style: [plugin.video.foo] v1.0.0

Addtional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuesly monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by romanvm** [comment]
My German is almost non-existent. What is the legal status of this site?

File: plugin.video.aod/addon.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.aod" name="Anime on Demand" version="0.4.13" provider-name="L0RE">
+    <requires>
+        <import addon="xbmc.python" version="2.1.0"/>
```
Comment (phil65) [inline plugin.video.aod/addon.xml]: 2.24.0

File: plugin.video.aod/changelog.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿0.1 
+- Initial Source
+0.2 
+- Bug Fix
```
Comment (phil65) [inline plugin.video.aod/changelog.txt]: please either translate the changelog to english or remove the changelog.

File: plugin.video.aod/default.py [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre})
+  liz.setProperty('IsPlayable', 'true')
+  liz.addStreamInfo('video', { 'duration' : duration })
+  liz.setProperty("fanart_image", iconimage)
+  #liz.setProperty("fanart_image", defaultBackground)
+  xbmcplugin.setContent(int(sys.argv[1]), 'tvshows')
+  ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz)
+  return ok
+  
+def all(url=""):
```
Comment (phil65) [inline plugin.video.aod/default.py]: better use another function name, "all()" is also a built-in function.

File: plugin.video.aod/resources/language/English/strings.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (phil65) [inline plugin.video.aod/resources/language/English/strings.xml]: could you convert the xmls to .po files, please? Language xml files are deprecated.

Comment (phil65) [inline plugin.video.aod/resources/language/English/strings.xml]: See https://github.com/alanwww1/xbmc-xml2po
https://github.com/alanwww1/xbmc-xml2po/downloads

File: plugin.video.aod/default.py [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+  if mode == 'All':
+          all(baseurl+"/animes")
+  if mode == 'Serie':
+          Serie(url) 
+  if mode == 'Folge':
+          Folge(url,csrftoken,type)            
+  if mode == 'cat':
+          category()  
+  if mode == 'lang':
+          lanuage()
```
Comment (phil65) [inline plugin.video.aod/default.py]: "language"

File: plugin.video.aod/default.py [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+  cj.save(cookie,ignore_discard=True, ignore_expires=True)
+
+  kurz_inhalt = content[content.find('<ul class="inline-block-list" style="display: block; text-align: center;">')+1:]                                      
+  kurz_inhalt = kurz_inhalt[:kurz_inhalt.find('</ul>')]  
+  match=re.compile('<a href="([^"]+)">([^<]+)</a>', re.DOTALL).findall(kurz_inhalt)
+  for link,name in match:
+      addDir(name=ersetze(name), url=baseurl+link, mode="catall",iconimage="", desc="")
+  xbmcplugin.endOfDirectory(addon_handle,succeeded=True,updateListing=False,cacheToDisc=True)
+ 
+def lanuage() :
```
Comment (phil65) [inline plugin.video.aod/default.py]: "language".

File: plugin.video.aod/resources/language/English/strings.xml (hunk 4 above)
Comment (MartijnKaijser) [inline plugin.video.aod/resources/language/English/strings.xml]: This is not needed immediately ofcourse. Just something you can do for new addons already.

File: plugin.video.aod/changelog.txt (hunk 2 above)
Comment (MartijnKaijser) [inline plugin.video.aod/changelog.txt]: For changelogs i personnaly don't really see the issue as this plugin is intended for Germans only. The language files are of course something different.  My advise though is that you put the latest changes on top in the changelog.
Also consider adding the <news> section in future updates as that will replace the changelog.txt
See http://forum.kodi.tv/showthread.php?tid=250936&pid=2361575#pid2361575  for some info.

**Comment by tyl0re** [comment]
Hopefully i have corrected all. Changed the Version to 0.4.14 because of the Changes. Im Not shure to make a new PR.

**Comment by phil65** [comment]
Changelog is still german.

**Comment by tyl0re** [comment]
Phil65:
Thought German doesnt matter,because of:

@MartijnKaijser commented on this pull request.
For changelogs i personnaly don't really see the issue as this plugin is intended for Germans only. The language files are of course something different. My advise though is that you put the latest changes on top in the changelog.
Also consider adding the section in future updates as that will replace the changelog.txt
See http://forum.kodi.tv/showthread.php?tid=250936&pid=2361575#pid2361575 for some info

**Comment by tamland** [comment]
Perhaps [\<language\>](http://kodi.wiki/view/Addon.xml#.3Clanguage.3E) should be set to de if it's german only?

**Comment by tyl0re** [comment]
Addad Laguage

File: plugin.video.aod/resources/language/German/strings.po [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Language: de\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\n"
+
+msgctxt "Addon Summary"
+msgid "Videos von http://www.anime-on-demand.de/"
```
Comment (enen92) [inline plugin.video.aod/resources/language/German/strings.po]: Think you made a mistake here :)

**Comment by tyl0re** [comment]
I got  an email

enen92 commented on this pull request.
+msgid ""
+msgstr ""
+"Project-Id-Version: XBMC-Addons\n"
+"Report-Msgid-Bugs-To: alanwww1@xbmc.org\n"
+"POT-Creation-Date: 2016-10-22 19:44+0000\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Language: de\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\n"
+
+msgctxt "Addon Summary"
+msgid "Videos von http://www.anime-on-demand.de/"

Think you made a mistake here :)

I dont Find the Command here. I asked Via Email what error you mean"
Since im Not shure the Repsonse got anywhere. I thought i Ask the Question again on the Missing Question here

**Comment by enen92** [comment]
@tyl0re please ignore the email, it was referring to a comment I deleted afterwards. Those strings "Addon Summary" and "Addon Description" are probably not being used, but there is no need for you to delete them. 
Either way, even if they were being used you have the german translation on the english .po file and the english translation on the german .po file for those streams.

**Comment by tyl0re** [comment]
Zitat von enen92 <notifications@github.com>:

> @ENEN92 commented on this pull request.
>
> -------------------------
>
>   In plugin.video.aod/resources/language/German/strings.po[1]:
>> +msgid "" +msgstr "" +"Project-Id-Version: XBMC-Addons\n"  
>> +"Report-Msgid-Bugs-To: alanwww1@xbmc.org\n" +"POT-Creation-Date:  
>> 2016-10-22 19:44+0000\n" +"PO-Revision-Date: YEAR-MO-DA  
>> HO:MI+ZONE\n" +"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"  
>> +"Language-Team: LANGUAGE\n" +"MIME-Version: 1.0\n" +"Content-Type:  
>> text/plain; charset=UTF-8\n" +"Content-Transfer-Encoding: 8bit\n"  
>> +"Language: de\n" +"Plural-Forms: nplurals=2; plural=(n != 1)\n" +  
>> +msgctxt "Addon Summary" +msgid "Videos von  
>> http://www.anime-on-demand.de/"
>   Think you made a mistake here :)
>
>   —

Whats the Misstake?


Links:
------
[1] https://github.com/xbmc/repo-plugins/pull/719#pullrequestreview-5366180

File: plugin.video.aod/addon.xml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="default.py">
+        <provides>video</provides>
+    </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Videos von http://www.anime-on-demand.de/</summary>
+    <summary lang="de">Videos of http://www.anime-on-demand.de/</summary>
+    <description lang="en">Videos der Mediathek von http://www.anime-on-demand.de/ (Die erste Folge ist Kostenlos. Ein Account wird benoetigt)</description>
+    <description lang="de">Video From http://www.anime-on-demand.de/ (Die First Episode is Free. Accound is Required)</description>
```
Comment (ronie) [inline plugin.video.aod/addon.xml]: "en" is in german and "de" is in english for summary/description ;-)

**Comment by tyl0re** [comment]
Im Not shure what gone wrong now its in

**Pull request merged**